src/pdf_surface.rs

```diff
@@ -0,0 +1,141 @@
+
+use std::ffi::CString;
+use std::ops::Deref;
+
+extern crate libc;
+use libc::{c_char, c_double};
+
+use ffi;
+use ffi::enums::SurfaceType;
+
+use surface::{Surface, SurfaceExt};
+
+#[cfg(feature = "glib")]
+use glib::translate::*;
+
+pub struct PDFSurface(Surface);
+
+
+extern "C" {
+    pub fn cairo_pdf_surface_create (filename: *const c_char,
+                                     width_in_points: c_double,
+                                     height_in_points: c_double) -> *mut ffi::cairo_surface_t;
+}
+
+impl PDFSurface {
+    pub fn from(surface: Surface) -> Result<PDFSurface, Surface> {
+        if surface.get_type() == SurfaceType::Pdf {
+            Ok(PDFSurface(surface))
+        } else {
+            Err(surface)
```

Comment:
We do same on ImageSurface and Win32Surface.
Normally error branch not work, but if it happened user get surface and can check it thoroughly